### PR TITLE
Fix sbt plugin incompatibilities on Windows

### PR DESCRIPTION
**Pull request by WojciechMazur**
This PR fixes fatal errors in the sbt plugin when trying to use it on Windows. Changes include: 
* detection of os used in sbt plugin
* using `where` command instead of `which` 
* passing paths in process commands in sequences, to correctly handle spaces
* fixes to regex and globing filters including usage of the correct path separator

Some methods with single usage were inlined. 

There are still some pending sbt plugin related issues that would need to resolve in the following PRs, e.g.:
* Handling `llvm-config` dependent settings - `llvm-config` is not included in prebuilt binaries targeted for Windows. We don't want to force users to build it locally. Probably for `include` and `lib` dir we can use fixed relative paths based on the location of `clang` or `clang++`.  I have such change prepared, however I decided to not include it now, because of next problem
* Adding proper handling and detection of `include` and `lib` directories, however might finally decide to leave this part for users to configure.

### CODE REVIEW

**Comment by ekrich** [comment]
I had put in an issue https://github.com/scala-native/scala-native/issues/2127 and had at least a plan sketch to use the `llvm-config` command to get compiling and linking options and only adding defaults if it didn't work. Using Mac tools instead of brew is strange because there is no installed `llvm-config` for example.

I found that `llvm-config` as it is now didn't work as it normally is `llvm-config-6` for example if version 6. I also found that `clang++` is normally aliased to `clang` so maybe it would make more sense to default to `CLANG_PATH` instead of the current. Maybe there is a use case to have a different version of C and C++ compilers but I am not sure.

I think PR right now looks good.

File: tools/src/main/scala/scala/scalanative/build/NativeLib.scala [hunk 1]
```diff
@@ -185,4 +178,12 @@ private[scalanative] object NativeLib {
       case true  => Some(path)
       case false => None
     }
+
+  private def makeDirPath(path: Path, elems: String*): String = {
+    val pathSep = if (Platform.isWindows) raw"\\" else File.separator
```
Comment (ekrich) [inline tools/src/main/scala/scala/scalanative/build/NativeLib.scala:183]: This is needed because it is inside a `glob` which is like a regex?

Comment (WojciechMazur) [reply tools/src/main/scala/scala/scalanative/build/NativeLib.scala:183]: Yes, I think that glob in java is also implemented based on regex. 
Without this, we would not be able to correctly find all native libs.

File: tools/src/main/scala/scala/scalanative/build/Discover.scala [hunk 2]
```diff
@@ -85,7 +85,8 @@ object Discover {
   private[scalanative] def checkClangVersion(pathToClangBinary: Path): Unit = {
     def versionMajorFull(clang: String): (Int, String) = {
       val versionCommand = s"$clang --version"
```
Comment (sjrd) [inline tools/src/main/scala/scala/scalanative/build/Discover.scala:87]: Unused?

File: tools/src/main/scala/scala/scalanative/build/Discover.scala [hunk 3]
```diff
@@ -133,8 +134,9 @@ object Discover {
    */
   private[scalanative] def discover(binaryName: String,
                                     envPath: String): Path = {
-    val binaryNameOrPath = sys.env.get(envPath).getOrElse(binaryName)
-    val path = Process(s"which $binaryNameOrPath")
+    val binaryNameOrPath = sys.env.getOrElse(envPath, binaryName)
+    val locateCmd        = if (Platform.isWindows) "where" else "which"
+    val path = Process(s"$locateCmd $binaryNameOrPath")
```
Comment (sjrd) [inline tools/src/main/scala/scala/scalanative/build/Discover.scala:139]: Should this also use a `Seq` for the arguments?

File: tools/src/main/scala/scala/scalanative/build/Platform.scala [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package scala.scalanative.build
+
+object Platform {
+  private lazy val osUsed = System.getProperty("os.name", "unknown").toLowerCase
```
Comment (sjrd) [inline tools/src/main/scala/scala/scalanative/build/Platform.scala:4]: This won't work on a Turkish/Lithuanian/Azeri system.
```suggestion
  private lazy val osUsed = System.getProperty("os.name", "unknown").toLowerCase(Locale.ROOT)
```

**Review by sjrd — CHANGES_REQUESTED** [review]

**Review by sjrd — APPROVED** [review]

**Pull request merged**